geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/netty/Coder.java

```diff
@@ -471,4 +472,101 @@ public static int narrowLongToInt(long toBeNarrowed) {
     }
     return doubleBytes;
   }
+
+  /**
+   * Takes the given "value" and computes the sequence of ASCII digits it represents,
+   * appending them to the given "buf".
+   * This code was adapted from the openjdk Long.java getChars methods.
+   */
+  public static void appendAsciiDigitsToByteBuf(long value, ByteBuf buf) {
+    if (value < 0) {
+      buf.writeByte('-');
+    } else {
+      value = -value;
+    }
+    // at this point value <= 0
+
+    if (value > -100) {
+      // it has at most two digits: [0..99]
+      appendSmallAsciiDigitsToByteBuf((int) value, buf);
+    } else {
+      appendLargeAsciiDigitsToByteBuf(value, buf);
+    }
+  }
+
+  private static void appendSmallAsciiDigitsToByteBuf(int value, ByteBuf buf) {
+    int q = value / 10;
+    int r = (q * 10) - value;
+    if (q < 0) {
+      buf.writeByte((byte) ('0' - q));
+    }
+    buf.writeByte((byte) ('0' + r));
```

Comment:
These zeroes can be replaced with the `NUMBER_0_BYTE` constant.

Reply:
done